### PR TITLE
chore: Replace React.FC with basic function syntax

### PR DESCRIPTION
**Pull request by BrightonMboya**
Remove React.FC usage in templates, replacing with standard function syntax as per Cje tweet https://twitter.com/ccccjjjjeeee/status/1668694515484336145

### CODE REVIEW

File: apps/expo/src/app/index.tsx [hunk 1]
```diff
@@ -118,7 +118,9 @@ const Index = () => {
           data={postQuery.data}
           estimatedItemSize={20}
           ItemSeparatorComponent={() => <View className="h-2" />}
-          renderItem={(p) => (
+          renderItem={(p: {
+            item: { id: string; content: string; title: string };
```
Comment (juliusmarminge) [inline apps/expo/src/app/index.tsx:122]: This should be inferred

File: apps/nextjs/src/pages/index.tsx [hunk 2]
```diff
@@ -110,15 +110,17 @@ const Home: NextPage = () => {
...
-                      return (
-                        <PostCard
-                          key={p.id}
-                          post={p}
-                          onPostDelete={() => deletePostMutation.mutate(p.id)}
-                        />
-                      );
-                    })}
+                    {postQuery.data?.map(
+                      (p: { id: string; title: string; content: string }) => {
```
Comment (juliusmarminge) [inline apps/nextjs/src/pages/index.tsx:114]: Again, should be inferred.

Did yii ok u forget to run db:generate? Might have to restart the TS server afterwards too

**Review by juliusmarminge — CHANGES_REQUESTED** [review]
Remove the explicit typings of the `p` variables

File: apps/nextjs/src/pages/index.tsx (hunk 2 above)
Comment (BrightonMboya) [reply apps/nextjs/src/pages/index.tsx]: I think I didn't restart the TS server thats why I kept on seeing the squiggly lines. I have removed it rn

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**